### PR TITLE
Added must-gather image

### PR DESCRIPTION
**Pull request by slintes**
It will gather
- objects in the operator's namespace
- nodemaintenance CRs
- nodes

```release-note
Added must-gather image
```

### CODE REVIEW

**Comment by slintes** [comment]
@beekhof ready for review. If you want to test it, follow the README, I pushed an image manually.

File: must-gather/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+You can use the `oc adm must-gather` command to collect information about your cluster.
+
+With the lifecycle must-gather image you can collect manifests and logs related to the nodes' lifecycle,
+which today includes the node objects, and logs and manifests related to the node-maintenance-operator.
+
+To collect this data, you must specify the extra image using the `--image` option.
+Example:
+
+```bash
+oc adm must-gather --image=quay.io/kubevirt/lifecycle-must-gather:latest
```
Comment (beekhof) [inline must-gather/README.md:12]: You had permissions to create a new image?

File: must-gather/collection-scripts/gather [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+[ -z "${NMO_NAMESPACE}" ] && ns="openshift-node-maintenance"
+
+# resource list
+resources=()
+
+# operator namespace
+resources+=(ns/${NMO_NAMESPACE})
+
+# node maintenance CRs
+resources+=(nodemaintenance)
```
Comment (beekhof) [inline must-gather/collection-scripts/gather:20]: I think we should also collect Nodes and Machines
Or are they automatically collected by some base level functionality?

**Pull request merged**